src/Stripe.Tests.XUnit/Services/Coupons/StripeCouponServiceTest.cs

```diff
@@ -0,0 +1,105 @@
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Stripe.Tests.Xunit
+{
+    public class StripeCouponServiceTest : BaseStripeTest
+    {
+        public const string CouponId = "co_123";
+
+        public StripeCouponService Service;
+        public StripeCouponCreateOptions CreateOptions;
+        public StripeCouponUpdateOptions UpdateOptions;
+        public StripeCouponListOptions ListOptions;
+
+        public StripeCouponServiceTest() {
+            Service = new StripeCouponService();
+
+            CreateOptions = new StripeCouponCreateOptions() {
+                PercentOff = 25,
+                Duration = "forever",
+            };
+
+            UpdateOptions = new StripeCouponUpdateOptions() {
+                Metadata = new Dictionary<string, string>() {
+                    { "key", "value" },
+                },
+            };
+
+            ListOptions = new StripeCouponListOptions() {
+                Limit = 1,
+            };
+        }
+
+        [Fact]
+        public void Create()
+        {
+            var coupon = Service.Create(CreateOptions);
+            Assert.NotNull(coupon);
+        }
+
+        [Fact]
+        public async Task CreateAsync()
+        {
+            var coupon = await Service.CreateAsync(CreateOptions);
+            Assert.NotNull(coupon);
+        }
+
+        [Fact]
+        public void Delete()
+        {
+            var deleted = Service.Delete(CouponId);
+            Assert.NotNull(deleted);
+        }
+
+        [Fact]
+        public async Task DeleteAsync()
+        {
+            var deleted = await Service.DeleteAsync(CouponId);
+            Assert.NotNull(deleted);
+        }
+
+        [Fact]
+        public void Get()
+        {
+            var coupon = Service.Get(CouponId);
+            Assert.NotNull(coupon);
+        }
+
+        [Fact]
+        public async Task GetAsync()
+        {
+            var coupon = await Service.GetAsync(CouponId);
+            Assert.NotNull(coupon);
+        }
+
+        [Fact]
+        public void List()
+        {
+            var coupons = Service.List(ListOptions);
+            Assert.NotNull(coupons);
```

Comment:
Should we confirm what's _in_ the list?

Reply:
Not sure there'd be much value in it since I think whatever stripe-mock returns will be coerced into the expected type. I'll check though.

Reply:
yeah you're likely right. I always have a hard time letting go of all the code that tested (and caught) so many thing before but it's likely superfluous and we just need good tests for `StripeList` in general